### PR TITLE
Make formats in `SaveFormatPanel` less error-prone

### PR DESCRIPTION
**Pull request by Floessie**
Hi,

This is the continuation of Ingo's work to fix the format list in `SaveFormatPanel`.

Best,
Flössie

### CODE REVIEW

File: rtgui/saveformatpanel.cc [hunk 1]
```diff
@@ -124,31 +124,29 @@ SaveFormatPanel::~SaveFormatPanel ()
...
-    } else if (sf.format == "tif" && sf.tiffBits == 16 && sf.tiffFloat) {
-        format->set_active (3);
-    } else if (sf.format == "tif" && sf.tiffBits == 16) {
-        format->set_active (2);
-    } else if (sf.format == "tif" && sf.tiffBits == 8) {
-        format->set_active (1);
+    std::pair<int, std::size_t> index;
+
+    for (std::size_t i = 0; i < sf_templates.size(); ++i) {
+        const int weight =
```
Comment (heckflosse) [inline rtgui/saveformatpanel.cc]: @Floessie The calculation of `weight` needs to be explained imho

**Review by heckflosse — COMMENTED** [review]
:+1: for merge

File: rtgui/saveformatpanel.cc (hunk 1 above)
Comment (Floessie) [reply rtgui/saveformatpanel.cc]: @heckflosse I'll add some comments to the code...

**Review by Beep6581 — APPROVED** [review]

File: rtgui/saveformatpanel.cc (hunk 1 above)
Comment (heckflosse) [reply rtgui/saveformatpanel.cc]: :+1:

**Pull request merged**